modules/addressing-cuda-oom-errors.adoc

```diff
@@ -0,0 +1,51 @@
+:_module-type: PROCEDURE
+
+[id="addressing-cuda-oom-errors-for-the-single-model-serving-platform_{context}"]
+= Addressing CUDA Out of Memory errors
+
+[role="_abstract"]
+
+In certain cases, depending on the model and hardware accelerator used, the TGIS memory auto tuning algorithm may underestimate the amount of GPU memory used to process long sequences. This can lead to CUDA OOM error responses from the model server. In such cases, we recommend updating or adding additional parameters in the TGIS model-serving runtime as described in the following procedure.
+
+
+.Prerequisites
+* You have logged in to {productname-long}.
+ifdef::upstream[]
+* If you are using specialized {productname-short} groups, you are part of the admin group (for example, {odh-admin-group}) in {openshift-platform}.
+endif::[]
+ifndef::upstream[]
+* If you are using specialized {productname-short} groups, you are part of the admin group (for example, {oai-admin-group}) in {openshift-platform}.
+endif::[]
+* You have installed KServe.
+* You have enabled a custom model-serving runtime for TGIS.
+
+
+.Procedure
+. From the {productname-short} dashboard, click *Settings* > *Serving runtimes*.
++
+The *Serving runtimes* page opens and shows the model-serving runtimes that are already installed and enabled.
++
+. Find your custom runtime for TGIS, click the action menu next to the runtime and select *Edit*.
++
+The embedded YAML editor opens and shows the contents of the custom runtime.
++
+. Add or update the `BATCH_SAFETY_MARGIN` environment variable and set the value to 30. Similarly, add or update the `ESTIMATE_MEMORY_BATCH_SIZE` environment variable and set the value to 8.
++
+[source]
+----
+spec:
+  containers:
+    env:
+    - name: BATCH_SAFETY_MARGIN
+      value: 30
+    - name: ESTIMATE_MEMORY_BATCH
+      value: 8
+----
++
+[NOTE]
+====
+The `BATCH_SAFETY_MARGIN` parameter sets a percentage of free GPU memory to hold back as a safety margin to avoid OOM conditions. The default value of `BATCH_SAFETY_MARGIN` is 20. The `ESTIMATE_MEMORY_BATCH_SIZE` parameter sets the batch size used in the memory auto-tuning algorithm. The default value of `ESTIMATE_MEMORY_BATCH_SIZE`  is 16.
+====
+. Click *Save*.
```

Comment:
What happens when you click Save? How does it affect models that are already running?

Reply:
Addressed in the latest commit.